tests/show_test.go

```diff
@@ -120,77 +116,32 @@ func TestShow(t *testing.T) {
 		assert.NotContains(t, out, "and")
 	})
 
-	// what if we disable the new secret type after having been using it?
-	t.Run("after disabling mime", func(t *testing.T) {
-		_, err = ts.run("config mime false")
-		assert.NoError(t, err)
-		_, err = ts.run("config safecontent true")
-		assert.NoError(t, err)
-
-		// No flags
-		out, err := ts.run("show fixed/secret")
-		assert.NoError(t, err)
-		assert.Contains(t, out, "***")
-		assert.NotContains(t, out, "Warning: safecontent=true")
-
-		out, err = ts.run("show fixed/twoliner password")
-		assert.NoError(t, err)
-		assert.Equal(t, out, "and")
-
-		out, err = ts.run("show fixed/twoliner")
-		assert.NoError(t, err)
-		assert.Contains(t, out, "more stuff")
-		assert.NotContains(t, out, "and")
-
-		// with flags now
-		out, err = ts.run("show -o fixed/secret")
-		assert.NoError(t, err)
-		assert.Equal(t, "moar", out)
-
-		out, err = ts.run("show -u fixed/secret")
-		assert.NoError(t, err)
-		assert.Equal(t, "moar", out)
-
-		out, err = ts.run("show -o fixed/twoliner")
-		assert.NoError(t, err)
-		assert.Equal(t, "and", out)
-
-		out, err = ts.run("show -u fixed/twoliner")
-		assert.NoError(t, err)
-		assert.Equal(t, "and\n\nmore stuff", out)
-	})
-
 	t.Run("Regression test for #1574 and #1575", func(t *testing.T) {
 		_, err = ts.run("config safecontent true")
 		assert.NoError(t, err)
-		_, err = ts.run("config mime false")
-		assert.NoError(t, err)
 
 		_, err := ts.run("generate fo2 5")
 		assert.NoError(t, err)
 
-		out, err := ts.run("show fo2")
-		assert.Error(t, err)
-		assert.Contains(t, out, "Warning: safecontent=true")
+		//out, err := ts.run("show fo2")
+		//assert.Error(t, err)
+		//assert.Contains(t, out, "Warning: safecontent=true")
 
-		out, err = ts.run("show -u fo2")
+		out, err := ts.run("show -u fo2")
 		assert.NoError(t, err)
 		assert.Equal(t, out, "aaaaa")
 
-		_, err = ts.run("config mime true")
-		assert.NoError(t, err)
-
 		_, err = ts.run("generate fo6 5")
 		assert.NoError(t, err)
 
 		out, err = ts.run("show fo6")
 		assert.NoError(t, err)
-		assert.Contains(t, out, "Password: ***")
+		assert.Contains(t, out, "password: ***")
```

Comment:
and this is the case only for KV secrets now, no?
Also we're missing a `assert.NotContains` the actual password, I guess.

Reply:
What do you mean?

Reply:
I mean that this test is testing that the `safecontent` option is correctly obstructing the output. And my point was that for plain secrets we should be doing any parsing, so it should only do so for KV values and instead for plain secrets the safecontent is just removing the first line, no?

Or do we replace the first line by `"password: ***"` in the case of plain secrets?

Reply:
Ah, good point.

For plain secrets we assume that the first line contains the secret, so we could support that there as well.